### PR TITLE
Implements QR Code refresh

### PR DESCRIPTION
**Pull request by kaykyr**

### CODE REVIEW

File: src/WAConnection/Validation.ts [hunk 1]
```diff
@@ -73,6 +74,15 @@ export default class WAConnectionValidator extends WAConnectionBase {
                 return this.userMetaData
             })
     }
+
+    /** Refresh QR Code */
+    protected refreshQrCode() {
+        const data = ['admin', 'Conn', 'reref']
+        return this.query(data).then((json) => {
```
Comment (adiwajshing) [inline src/WAConnection/Validation.ts:81]: Put this in a single line?

File: src/WAConnection/Validation.ts [hunk 2]
```diff
@@ -152,13 +162,24 @@ export default class WAConnectionValidator extends WAConnectionBase {
...
             this.authInfo.clientID,
         ])
-        return this.waitForMessage('s1', [])
+
+        try {
+            const response = await this.waitForMessage('s1', [], 20 * 1000)
+            return response
+        } catch (err) {
+            if (this.loops < 5) {
+                this.loops++
```
Comment (adiwajshing) [inline src/WAConnection/Validation.ts:178]: Any particular reason you made `loops` a class variable and not a local functional variable?

File: src/WAConnection/Validation.ts [hunk 3]
```diff
@@ -152,13 +162,24 @@ export default class WAConnectionValidator extends WAConnectionBase {
...
         this.curveKeys = Curve.generateKeyPair(Utils.randomBytes(32))
         this.onReadyForPhoneAuthentication([
             ref,
             Buffer.from(this.curveKeys.public).toString('base64'),
             this.authInfo.clientID,
         ])
-        return this.waitForMessage('s1', [])
+
+        try {
+            const response = await this.waitForMessage('s1', [], 20 * 1000)
```
Comment (adiwajshing) [inline src/WAConnection/Validation.ts:174]: Also, I'm fairly certain the message tag (`s1`) will change after a re-ref, so waiting for that message will never end. 

Does the tag not change?

Comment (kaykyr) [reply src/WAConnection/Validation.ts:174]: This is to wait for QR Code scanning confirmation, so while confirmation is not ready the script will keep refreshing QR Code until this message response, or until max loops reached.

File: src/WAConnection/Validation.ts (hunk 2 above)
Comment (kaykyr) [reply src/WAConnection/Validation.ts:178]: Sorry, my bad. I'll resend pull request without as class variable.

File: src/WAConnection/Validation.ts (hunk 1 above)
Comment (kaykyr) [reply src/WAConnection/Validation.ts:81]: It's better to keep clean code.

**Pull request closed without merging**